addons/html_editor/static/tests/_helpers/selection.js

```diff
@@ -214,3 +216,42 @@ function visitAndSetRange(target, ref, configSelection) {
         }
     }
 }
+
+export async function firstClick(target) {
+    manuallyDispatchProgrammaticEvent(target, "mousedown", { detail: 1 });
+    setSelection({ anchorNode: target, anchorOffset: 0 });
+    await animationFrame(); // selectionChange
```

Comment:
We really need a util to wait for the next selectionChange, don't we? cc @dmo-odoo